client/app/components/blueprints/blueprint-details-modal/blueprint-details-modal-service.factory.js

```diff
@@ -260,35 +278,11 @@
       }
 
       if (action === 'publish') {
-        $uibModalInstance.close();
-        saveFailure();
-
-        return;
+        BlueprintsState.publishBlueprint(vm.blueprint).then(saveSuccess, publicationFailure);
```

Comment:
Action not required: consider using names `success` and `failure`, collisions avoided with similar functions in a controller by declaring  aforementioned functions in the parent function